crates/sui-storage/src/execution_cache.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use sui_types::{
+    digests::{TransactionDigest, TransactionEffectsDigest},
+    error::SuiResult,
+    storage::ObjectKey,
+};
+
+/// The result of reading an object for execution. Because shared objects may be deleted, one
+/// possible result of calling ExecutionCache::notify_read_objects_for_execution on a shared object
+/// is that ObjectReadResult::Deleted is returned.
+pub enum ObjectReadResult {
+    Object(Object),
+    Deleted,
+}
+
+/// ExecutionCache is intended to provide an in-memory, write-behind or write-through cache used
+/// by transaction signing and execution.
+///
+/// Note that except where specified below, we do not provide any durability guarantees.
+/// Crash recovery is done by re-execution of transactions when necessary.
+#[async_trait]
+pub trait ExecutionCache {
```

Comment:
Is this an outdated version of the cache? It doesn't seem to be used anywhere. IIUC, the InMemoryCache with ExecutionCacheRead/ExecutionCacheWrite traits are the current execution cache implementation?

Reply:
yes, i need to delete this file, thanks for the reminder